### ISSUE TITLE
Incomplete python wrapping of enum class objects

### ISSUE THREAD

**Issue by mseng10** [InsightSoftwareConsortium/ITKExamples#131]
### Description

In the most recent set of failing ITKExamples tests, #130, the listed tests are unsolvable unless changed from the source. In the latest release of the ITKPythonPackage 5.1b.1 errors occur when attempting to access python wrapped enumerations. 

The issue arises because some enumerations are no longer wrapped after the 
#1061 pull request (In ITK). In #1061, they were changed from 'c' style enumerations to enum classes. During this transition a number of new enums were moved outside their original classes.

### Steps to Reproduce

1. Clone the latest version of ITKExamples from Github: 

`git clone -b master https://github.com/InsightSoftwareConsortium/ITKExamples.git`

3. Make build directory:  `mkdir ITKExamples-Superbuild`

4. Allocate to build directory:  `cd ITKExamples-Superbuild`

5. Run:  `cmake ../ITKExamples/Superbuild`

6. Configure CMake with:  `ITKExamples_USE_WRAP_PYTHON = ON`

7. Run: `make`

8. Allocate to ITKEx-build directory: `cd ITKEx-build`

9. Test: `ctest -V`
 

### Expected behavior

100% Tests passed.

### Actual behavior

```   
98% tests passed, 12 tests failed out of 589
248 - ApplyAColormapToAnImageTestBaselineComparisonPython (Failed)
252 - ApplyAColormapToAnImageTestPython (Failed)
568 - SegmentWithWatershedImageFilterTest01BaselineComparisonPython (Failed)
570 - SegmentWithWatershedImageFilterTest02BaselineComparisonPython (Failed)
572 - SegmentWithWatershedImageFilterTest03BaselineComparisonPython (Failed)
574 - SegmentWithWatershedImageFilterTest04BaselineComparisonPython (Failed)
576 - SegmentWithWatershedImageFilterTest05BaselineComparisonPython (Failed)
583 - SegmentWithWatershedImageFilterTest01Python (Failed)
584 - SegmentWithWatershedImageFilterTest02Python (Failed)
585 - SegmentWithWatershedImageFilterTest03Python (Failed)
586 - SegmentWithWatershedImageFilterTest04Python (Failed)
587 - SegmentWithWatershedImageFilterTest05Python (Failed)
```

### Reproducibility
The tests fail 100% of the time when the ITKPythonPackage 5.1b.1 is used. 

When ITKPythonPackage 5.0.1 is used, 100% of tests pass.

### Versions and Environment
- Done on macOS

- Python 3.7.3

- Latest ITKExamples, ITK, versions pulled from Github.

- ITKPythonPackage version **5.1b.1**

### Possible Solution
Lets examine an enumeration not wrapped: `RGBColormapFilterEnumType`

Path: /ITK/Modules/Filtering/Colormap/include/itkScalarToRGBColormapImageFilter.h

In the previous instance of this class, the class contained an instance variable that was a c-style enum. After being changed to a enum class, the new enum was placed outside the class (Line: 30). We can see the difference of the 2 classes when built as a python package.

#### ITKPythonPackage 5.0.1 (older)
Download the python package separate of your ITKExamples-Superbuild.

Allocate to: itk/itkScalarToRGBColormapImageFilterPython.py

As you can see on lines: 280-293, the enum values are specifically stated:
 (280)`Red=_itkScalarToRGBColormapImageFilterPython.itkScalarToRGBColormapImageFilterIF2IRGBAUC2_Red`

#### ITKPythonPackage 5.b1.1 (new)
Allocate to ITKExamples-Superbuild/itkpython/lib/python3.7/site-packages/itk/itkScalarToRGBColormapImageFilterPython.py

When inspecting the file, NO enum values are set, however it could be located outside the class .py file since it was moved out in the Source ITK.

When running the following steps to see if `RGBColormapFilterEnumType` is an attribute of itk we can see NO attribute RGBColormapFilterEnumType is listed.

###### Steps:

- Enter in terminal:  `python` (opens up python interface)

- Enter: ` import itk`

-  Enter: `dir(itk)`

The output will be a list of all attributes and methods for itk. As expected RGBColormapFilterEnumType is not listed, hence not wrapped.

#### Reason for not being wrapped
This is a bit past my understanding of the full architecture surrounding itk, but in the .py file: ITK/Wrapping/Generators/SwigInterface/igenerator.py we have a list of the enumerations to be wrapped.

On Line: 53, we see a list of enumerations set not to be wrapped and as expected RGBColormapFilterEnumType and many others are listed. I suspect modifying this file to begin wrapping enumerations listed from lines 90-98 will solve the issue. With these changes, users will be able to use the latest ITKPythonPackage with all of its functionality.  Please let me know of any questions as this is a larger issue with many components.

@hjmjohnson

**Comment by thewtex** [comment]
@mseng10 well done on the detailed and well-researched description!

Until we have the enum's wrapping in ITK's 5.1 Python wrapping, I propose we do not install the pre-release Python packages in the Superbuild here:

  https://github.com/InsightSoftwareConsortium/ITKExamples/pull/146

**Comment by hjmjohnson** [comment]
@thewtex  Did # 1353 address this completely?

**Comment by thewtex** [comment]
#1353 addressed the enum issues. We can get the tests passing after 5.1b2 binaries are published.

**Comment by mseng10** [comment]
Fixed in #156. Closing now.
